scripts/remove_unnecessary_versions_of_repositories.py

```diff
@@ -0,0 +1,135 @@
+# After changing the retain_repo_versions Pulp will remove unrelated versions
+async def remove_unnecessery_versions(
+    pulp_client: PulpClient,
+    min_retain_repo_versions: int,
+    pulp_repo: dict,
+    logger: logging.Logger,
+):
+    endpoint = pulp_repo["pulp_href"]
+    params = {"retain_repo_versions": min_retain_repo_versions}
+    result = await pulp_client.request("PATCH", endpoint, json=params)
+    logger.debug(
+        "Create task to change retain repo versions to %s",
+        params["retain_repo_versions"],
+    )
+    logger.debug(result)
+    params = {"retain_repo_versions": pulp_repo["retain_repo_versions"]}
+    result = await pulp_client.request("PATCH", endpoint, json=params)
+    logger.debug(
+        "Create task to back retain repo versions to %s",
+        params["retain_repo_versions"]
+    )
+    logger.debug(result)
+
+
+async def get_all_pulp_repositories(
+    pulp_client: PulpClient,
+    logger: logging.Logger,
+):
+    endpoint = "pulp/api/v3/repositories/rpm/rpm/"
+    params = {"fields": "id,pulp_href,retain_repo_versions"}
+    result = []
+
+    def update_result(response: typing.List[dict]):
+        for rec in response:
+            result.append(rec)
+
+    response = await pulp_client.request("GET", endpoint, params=params)
+    update_result(response.get("results", []))
+    next_page = response.get("next")
+    if not next_page:
+        return result
+    while True:
+        if (
+            "limit" in next_page
+            and re.search(r"limit=(\d+)", next_page).groups()[0] == "100"
+        ):
+            next_page = next_page.replace("limit=100", "limit=1000")
+        parsed_url = urllib.parse.urlsplit(next_page)
+        path = parsed_url.path + "?" + parsed_url.query
+        response = await pulp_client.get_by_href(path)
+        update_result(response.get("results", []))
+        next_page = response.get("next")
+        if not next_page:
+            break
+    logger.debug("%s repositories received from Pulp", len(result))
+    return result
+
+
+async def main():
+    pulp_host = os.environ["PULP_HOST"]
+    pulp_user = os.environ["PULP_USER"]
+    pulp_password = os.environ["PULP_PASSWORD"]
+    args = parse_args()
+    logger = logging.getLogger("pulp_artifacts_remover")
+    if args.verbose:
+        logging.basicConfig(level=logging.DEBUG)
+    else:
+        logging.basicConfig(level=logging.INFO)
+    pulp_client = PulpClient(pulp_host, pulp_user, pulp_password)
+    logger.info("Get all pulp repos")
+    pulp_repos = await get_all_pulp_repositories(pulp_client, logger)
+    logger.info("Removing unnecessery versions of repositories...")
+    for pulp_repo in pulp_repos:
+        await remove_unnecessery_versions(
+            pulp_client=pulp_client,
+            min_retain_repo_versions=3,
+            pulp_repo=pulp_repo,
+            logger=logger,
+        )
+    logger.info("All unnecessery version of repositories is deleted")
```

Comment:
"All unnecessary versions of repositories have been deleted"
Also, replace "unnecessery" with "unnecessary", found some of them